tools/eslint-rules/only-ascii-characters.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Prohibit the use of non-ascii characters
+ * @author Kalon Hinds
+ */
+
+/* eslint no-control-regex:0 */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const nonAsciiPattern = new RegExp('([^\x00-\x7F])', 'g');
+const suggestions = {
+  '’': '\'',
+  '—': '-'
+};
+const reportError = ({line, column, character}, node, context) => {
+  const suggestion = suggestions[character];
+
+  let message = `Non-ASCII character ${character} detected.`;
+
+  message = suggestion ?
+    `${message} Consider replacing with: ${suggestion}` : message;
+
+  context.report({
+    node,
+    message,
+    loc: {
+      line,
+      column
+    }
+  });
+};
+
+module.exports = {
+  create: (context) => {
+    return {
+      Program: (node) => {
+        const source = context.getSourceCode();
+        const sourceTokens = source.getTokens(node);
+        const commentTokens = source.getAllComments();
+        const tokens = sourceTokens.concat(commentTokens);
+
+        tokens.forEach((token) => {
```

Comment:
@aqrln by:

>If the latter is true

I mean that if:

> I *think* the [`no-irregular-whitespace`](http://eslint.org/docs/rules/no-irregular-whitespace) rule will cover non-ascii whitespace.

@not-an-aardvark's theory is correct, and the non-ASCII whitespace characters are already covered in a separate rule, then we could just use that for whitespace, and add a comment in here to explain that we don't need to worry about whitespace as it's covered in another rule.